src/main/java/com/patchworkmc/transformer/PatchworkTransformer.java

```diff
@@ -183,6 +188,14 @@ public void accept(String name, byte[] content) {
 		}
 
 		outputConsumer.accept(name, writer.toByteArray());
+
+		checker.onClassScanned(
+				name, subscribeEvents,
+				Stream.concat(
+						Stream.of(reader.getSuperName()),
+						Stream.of(reader.getInterfaces())
+				).collect(Collectors.toList())
+		);
```

Comment:
This is a good example of where I think using functional programming isn't appropriate. The straightforward imperative approach is less code than the functional approach, and in my opinion is much easier to understand.
```suggestion
		List<String> supers = new ArrayList<>();

		supers.add(reader.getSuperName());
		supers.addAll(reader.getInterfaces());

		checker.onClassScanned(name, subscribeEvents, supers);
```